Packs/CortexResponseAndRemediation/Playbooks/playbook-CVE-2025-49704_and_CVE-2025-49706_and_CVE-2025-53770_and_CVE-2025-53771_-_Microsoft_SharePoint_ToolShell_vulnerability_chain.yml

```
@@ -0,0 +1,3317 @@
+
+        #### List of IPs:
+        ${ExtractedIndicators.IP=>val.join(',')}
+
+        #### List of Hashes:
+        ${ExtractedIndicators.File=>val.join(',')}
+
+
+
+        XQL Query:
+        ```
+        config timeframe = 1h case_sensitive = false | datamodel dataset=* | FILTER   XDM_ALIAS.ip in(${ExtractedIndicators.IP=>val.join(',')}) or  ( XDM_ALIAS.file_hash in(${ExtractedIndicators.File=>val.join(',')}) | fields fieldset.xdm_core, xdm.intermediate.ipv4, xdm.source.host.ipv4_addresses, xdm.target.host.ipv4_addresses, xdm.intermediate.host.ipv4_addresses, xdm.network.dhcp.ciaddr, xdm.network.dhcp.yiaddr, xdm.network.dhcp.siaddr, xdm.network.dhcp.giaddr, xdm.network.vpn.allocated_ipv4, xdm.source.ipv6, xdm.target.ipv6, xdm.intermediate.ipv6, xdm.source.host.ipv6_addresses, xdm.target.host.ipv6_addresses, xdm.intermediate.host.ipv6_addresses, xdm.network.vpn.allocated_ipv6, xdm.source.process.executable.md5, xdm.source.process.executable.sha256, xdm.target.process.executable.md5, xdm.target.process.executable.sha256, xdm.intermediate.process.executable.md5, xdm.intermediate.process.executable.sha256, xdm.target.module.md5, xdm.target.module.sha256, xdm.target.file.md5, xdm.target.file.sha256, xdm.target.file_before.md5, xdm.target.file_before.sha256
+        ```
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "28"
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 2965,
+          "y": 3119
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "123":
+    id: "123"
+    taskid: b9c0cea4-0cc6-4f3f-92d8-9168e58d482e
+    type: regular
+    task:
+      id: b9c0cea4-0cc6-4f3f-92d8-9168e58d482e
+      version: -1
+      name: Manual – Check for exploitation in your environment
+      description: "Dear Analyst,\n\nEnsure that the 'XQL Query Engine' integration is active. If it's not enabled, activate the integration or manually run the following XQL query to determine if any IOCs are found within your environment. Otherwise, you can go ahead and search manually for the following indicators\n\n#### List of IPs:\n${ExtractedIndicators.IP=>val.join(',')}\n\n#### List of Hashes:\n${ExtractedIndicators.File=>val.join(',')}\n\n\n\nXQL Query:\n```\nconfig timeframe = 7d case_sensitive = false \n| dataset = *\n| filter (http_method = \"POST\" and http_req_uri contains \"/_layouts/*/ToolPane.aspx\")\nor (http_method = \"GET\" and http_req_uri contains \"/_layouts/*/spinstall0.aspx\")\n| alter exploit_type = if(http_referer contains \"/_layouts/SignOut.aspx\", \"RCE and unauthenticated\" , \"RCE\")\n| fields agent_hostname, http_referer, http_method, actor_process_image_name, _time  ,http_req_uri\n| sort desc _time \n```\n"
```

```suggestion
      description: "Dear Analyst,\n\nEnsure that the 'XQL Query Engine' integration is active. If it's not enabled, activate the integration or manually run the following XQL query to determine if any IOCs are found within your environment. Otherwise, you can go ahead and search manually for the following indicators.\n\n#### List of IPs:\n${ExtractedIndicators.IP=>val.join(',')}\n\n#### List of Hashes:\n${ExtractedIndicators.File=>val.join(',')}\n\n\n\nXQL Query:\n```\nconfig timeframe = 7d case_sensitive = false \n| dataset = *\n| filter (http_method = \"POST\" and http_req_uri contains \"/_layouts/*/ToolPane.aspx\")\nor (http_method = \"GET\" and http_req_uri contains \"/_layouts/*/spinstall0.aspx\")\n| alter exploit_type = if(http_referer contains \"/_layouts/SignOut.aspx\", \"RCE and unauthenticated\" , \"RCE\")\n| fields agent_hostname, http_referer, http_method, actor_process_image_name, _time  ,http_req_uri\n| sort desc _time \n```\n"
```